symfony/plugins/orangehrmAdminPlugin/Api/WorkShiftEmployeeAPI.php

```diff
@@ -0,0 +1,255 @@
+
+    public const MODEL_DEFAULT = 'default';
+    public const MODEL_DETAILED = 'detailed';
+    public const MODEL_MAP = [
+        self::MODEL_DEFAULT => EmployeeModel::class,
+        self::MODEL_DETAILED => EmployeeDetailedModel::class,
+    ];
+
+    public function getAll(): EndpointResult
+    {
+        $employeeParamHolder = new EmployeeSearchFilterParams();
+        $this->setSortingAndPaginationParams($employeeParamHolder);
+
+        $empNumber = $this->getRequestParams()->getIntOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::FILTER_EMP_NUMBER
+        );
+        if (!is_null($empNumber)) {
+            $employeeParamHolder->setEmployeeNumbers([$empNumber]);
+        } else {
+            $accessibleEmpNumbers = $this->getUserRoleManager()->getAccessibleEntityIds(Employee::class);
+            $employeeParamHolder->setEmployeeNumbers($accessibleEmpNumbers);
+        }
+
+        $employeeParamHolder->setIncludeEmployees(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_INCLUDE_EMPLOYEES
+            )
+        );
+        $employeeParamHolder->setName(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_NAME
+            )
+        );
+        $employeeParamHolder->setNameOrId(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_NAME_OR_ID
+            )
+        );
+        $employeeParamHolder->setEmployeeId(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_EMPLOYEE_ID
+            )
+        );
+        $employeeParamHolder->setEmpStatusId(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_EMP_STATUS_ID
+            )
+        );
+        $employeeParamHolder->setJobTitleId(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_JOB_TITLE_ID
+            )
+        );
+        $employeeParamHolder->setSubunitId(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_SUBUNIT_ID
+            )
+        );
+        $employeeParamHolder->setSupervisorEmpNumbers(
+            $this->getRequestParams()->getArrayOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_SUPERVISOR_EMP_NUMBERS
+            )
+        );
+        $count = $this->getEmployeeService()->getEmployeeCount($employeeParamHolder);
+        $employee = $this->getEmployeeService()->getAvailableEmployeesForWorkShift($employeeParamHolder);
+        return new EndpointCollectionResult(
+            $this->getModelClass(),
```

Comment:
no need to support multiple models, use only `EmployeeModel::class`

Reply:
done